File: Harvest/Models/Message.swift
import Foundation

struct Message: Codable, Identifiable, Sendable {
    let id: String
    let conversationId: String
    let senderId: String
    var content: String?
    var messageType: String?
    var mediaUrl: String?
    var isRead: Bool
    var readAt: String?
    let createdAt: String?

    enum CodingKeys: String, CodingKey {
        case id, content
        case conversationId = "conversation_id"
        case senderId = "sender_id"
        case messageType = "message_type"
        case mediaUrl = "media_url"
        case isRead = "is_read"
        case readAt = "read_at"
        case createdAt = "created_at"
    }

    func isSentBy(_ userId: String) -> Bool {
        senderId.lowercased() == userId.lowercased()
    }
}

File: android/app/src/main/java/com/harvestglass/harvest/data/model/Message.kt
package com.harvestglass.harvest.data.model

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

/** Mirrors Harvest/Models/Message.swift — a 1:1 conversation message. */
@Serializable
data class Message(
    val id: String,
    @SerialName("conversation_id") val conversationId: String,
    @SerialName("sender_id") val senderId: String,
    val content: String? = null,
    @SerialName("message_type") val messageType: String? = null,
    @SerialName("media_url") val mediaUrl: String? = null,
    @SerialName("is_read") val isRead: Boolean = false,
    @SerialName("read_at") val readAt: String? = null,
    @SerialName("created_at") val createdAt: String? = null
) {
    /** Case-insensitive, matching the Swift original. */
    fun isSentBy(userId: String): Boolean = senderId.lowercase() == userId.lowercase()
}
